docs/programming-guide.md

```diff
@@ -1022,6 +1022,27 @@ for details.
 </tr>
 </table>
 
+### Shuffle operations
+
+Certain operations within Spark trigger an operation known as the shuffle. The shuffle is Spark's mechanism for re-organizing data to co-locate data associated with particular keys. 
```

Comment:
maybe something like this?

"shuffle is Spark's mechanism for redistributing data so data of the same key becomes co-located after shuffle".